CHANGELOG.md

```diff
@@ -1,5 +1,20 @@
 :warning: **Important**: If you are upgrading to version 2.3.0 or later and have firewall rules or network configuration that blocks any unknown traffic by default, you need to update your configuration to allow connections to the new DNS names and IP addresses. Please refer to this [changelog](#230-january-23-2023) for more details.
 
+2.8.0 (In Progress)
+===================
+
+Changes
+-------
+
+- Enhanced error reporting from backend services. Signaling errors emitted by `Device` and `Call` objects should now be more descriptive. If your application logic relied on parsing generic error code `31005`, please consider reading through our API documentation and adjusting your application logic to anticipate a wider and more descriptive range of error codes.
+  - Some new error codes included within this release are:
+    - **ClientErrors**
+      - NotFound: `31404`
+      - TemporarilyUnavailable: `31480`
+      - BusyHere: `31486`
+    - **SIPServerErrors**
+      - Decline: `31603`
```

Comment:
Are all these new errors surfaced via both Call and Device objects?

Reply:
From my testing, no. Just the Call object since the Call object listens to hangup events, not error events. And the new error codes are associated with hangups and as such are only included in hangup type payloads.

Reply:
In that case, we should update the changelog and don't mention `Device`, since these new errors are only emitted by the `Call` object.

Reply:
So the thing is, I haven't been able to test `31404` and `31603`, so I'm not sure if the backend sends those as `errors` or as `hangups`. In the first case, if the pstream message type is `error`, then the device would show it, but if it's a `hangup`-associated error, then only the call will show it. I took a look at the voice signaling protocol repo and it doesn't say which error codes can show up in which types of payloads.